java/vortex-spark/src/main/java/dev/vortex/spark/read/VortexArrowColumnVector.java

```diff
@@ -0,0 +1,630 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
IANAL, but I think we gotta preserve Spark's license block. I also think we need a notice in the file that we made changes to this file.

Apache License section 4 parts b and c:

> b. You must cause any modified files to carry prominent notices stating that You changed the files; and
> c. You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works; and

In particular I think we might need to preserve:

> Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.

Reply:
yep i will change this